pkg/api/folder_test.go

```diff
@@ -378,6 +380,14 @@ func TestFolderMoveAPIEndpoint(t *testing.T) {
 				{Action: dashboards.ActionFoldersWrite, Scope: dashboards.ScopeFoldersProvider.GetResourceScopeUID("newParentUid")},
 			},
 		},
+		{
+			description:  "can move folder to the root folder with specific permissions",
+			newParentUid: "",
+			expectedCode: http.StatusOK,
+			permissions: []accesscontrol.Permission{
+				{Action: dashboards.ActionFoldersWrite, Scope: dashboards.ScopeFoldersProvider.GetResourceScopeUID("uid")},
+			},
+		},
```

Comment:
We actually also require the permission to create a folder to be able to move folder to the root level. This check is done in the folder service: https://github.com/grafana/grafana/blob/main/pkg/services/folder/folderimpl/folder.go#L494-L509

So I think this is not the right place to test for this. In general, it's not great that access control is scattered in several places, and it's something that we want to clean up in the future.

Reply:
ah I see: we use a fake service there